packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,410 @@
+ * ```
+ */
+export class MessagePrompter {
+	/**
+	 * The message that will been sent in [[MessagePrompter.run]]
+	 */
+	public message: IMessagePrompterMessage;
+
+	/**
+	 * The message that has been sent in [[MessagePrompter.run]]
+	 */
+	public appliedMessage: Message | null = null;
+
+	/**
+	 * The used options for this instance of [[MessagePrompter]]
+	 */
+	public options: IMessagePrompterOptions;
+
+	/**
+	 * The default options used if there is no type supplied.
+	 */
+	private defaultOptions: IMessagePrompterOptions = {
+		type: 'confirm',
+		reactions: {
+			confirm: '🇾',
+			cancel: '🇳'
+		},
+		timeout: 20 * 1000,
+		explicitReturn: false
+	};
+
+	/**
+	 * Constructor for the [[MessagePrompter]] class
+	 * @param message The message to send.
+	 * @param options Overrideable options if needed.
+	 */
+	public constructor(message: IMessagePrompterMessage, options?: Partial<IMessagePrompterOptions> | IMessagePrompterOptionsType) {
+		this.message = message;
+		const type: IMessagePrompterOptionsType = typeof options === 'string' ? options : options?.type ?? 'confirm';
+
+		this.options = {
+			...this.defaultOptions,
+			...(MessagePrompter.defaultOptions.get(type) ?? {}),
+			...((typeof options !== 'string' && options) ?? {})
+		};
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter The author to validate.
+	 */
+	public run(channel: TextChannel | NewsChannel | DMChannel, authorOrFilter: User | CollectorFilter): Promise<IMessagePrompterReturn> {
+		switch (this.options.type) {
+			case 'confirm':
+				return this.runConfirm(channel, authorOrFilter);
+			case 'number':
+				return this.runNumber(channel, authorOrFilter);
+			case 'reaction':
+				return this.runReaction(channel, authorOrFilter);
+			case 'message':
+				return this.runMessage(channel, authorOrFilter);
+		}
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message if [[IMessagePrompterOptions.type]] equals confirm.
+	 * The handler will wait for one (1) reaction.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter The author to validate.
+	 * @returns A promise that resolves to a boolean denoting the truth value of the input (`true` for yes, `false` for no).
+	 */
+	public async runConfirm(
+		channel: TextChannel | NewsChannel | DMChannel,
+		authorOrFilter: User | CollectorFilter
+	): Promise<IMessagePrompterExplicitReturn | boolean> {
+		const { reactions, timeout, explicitReturn } = this.options;
+		const { confirm, cancel } = reactions as IMessagePrompterOptionsConfirm;
+
+		const confirmReactions: string[] | EmojiIdentifierResolvable[] = [confirm, cancel];
+		const response = await this.collectReactions(channel, authorOrFilter, confirmReactions, timeout);
+
+		const confirmed = (response?.emoji?.id ?? response?.emoji?.name) === confirm;
+
+		return explicitReturn
+			? {
+					...response,
+					confirmed
+			  }
+			: confirmed;
```

Comment:
```suggestion
		return explicitReturn ? {...response, confirmed} : confirmed;
```
Way too typographic for such a short simple statement imo (and follow suit in other functions if you make this change)

Reply:
That's my prettier doing that

Reply:
```suggestion
		// prettier-ignore
		return explicitReturn ? {...response, confirmed} : confirmed;
```
Prettier be Prettier